tests/widgets/drive/drive-scanner.spec.coffee

```diff
@@ -0,0 +1,131 @@
+			removeSpy = m.sinon.spy()
+			m.chai.expect(removeSpy).to.not.have.been.called
+			scanner.on 'add', (drive) ->
+				m.chai.expect(scanner.drives).to.deep.equal([ {foo: 'bar'} ])
+				m.chai.expect(drive).to.deep.equal(foo: 'bar')
+				scanner.stop()
+				done()
+
+	describe 'given a driveFinder that finds one drive that has been removed', ->
+
+		beforeEach ->
+			@driveFinder = m.sinon.stub()
+			@driveFinder.returns(Promise.resolve([]))
+
+		it 'should emit a single remove event', (done) ->
+			scanner = new DriveScanner(@driveFinder, interval: 100, drives: [ {foo: 'bar'} ])
+			addSpy = m.sinon.spy()
+			m.chai.expect(addSpy).to.not.have.been.called
+			scanner.on 'remove', (drive) ->
+				m.chai.expect(scanner.drives).to.deep.equal([])
+				m.chai.expect(drive).to.deep.equal(foo: 'bar')
+				scanner.stop()
+				done()
+
+	describe 'given a driveFinder that finds one drive that has been changed', ->
+
+		beforeEach ->
+			@driveFinder = m.sinon.stub()
+			@driveFinder.onFirstCall().returns(Promise.resolve([ {foo: 'bar'} ]))
+			@driveFinder.returns(Promise.resolve([{foo: 'baz'}]))
+
+		it 'should emit a remote and an add event', (done) ->
+			scanner = new DriveScanner(@driveFinder, interval: 100)
+			removeSpy = m.sinon.spy()
+			scanner.on('remove', removeSpy)
+			addSpy = m.sinon.spy()
+			scanner.on('add', addSpy)
+			setTimeout ->
+				m.chai.expect(removeSpy).to.have.been.calledWith(foo: 'bar')
+				m.chai.expect(addSpy).to.have.been.calledWith(foo: 'baz')
+				scanner.stop()
+				done()
+			, 500
+
+	describe 'given that nothing happens', ->
+
+		beforeEach ->
+			@driveFinder = ->
+				return Promise.resolve([])
+
+		it 'should do nothing when no change has happened', (done) ->
+			scanner = new DriveScanner(@driveFinder, interval: 100)
+			removeSpy = m.sinon.spy()
+			scanner.on('remove', removeSpy)
+			addSpy = m.sinon.spy()
+			scanner.on('add', addSpy)
+			setTimeout ->
+				m.chai.expect(removeSpy).to.not.have.been.called
+				m.chai.expect(addSpy).to.not.have.been.called
+				scanner.stop()
+				done()
+			, 500
+
+	describe 'given two additions that happen at the same time', ->
+
+		beforeEach ->
+			@driveFinder = ->
+				return Promise.resolve([ {foo: 'bar'}, {foo: 'baz'} ])
+
+		it 'should emit two add events', (done) ->
+			scanner = new DriveScanner(@driveFinder, interval: 100)
+			addSpy = m.sinon.spy()
+			scanner.on('add', addSpy)
+			removeSpy = m.sinon.spy()
+			scanner.on('remove', removeSpy)
+			setTimeout ->
+				m.chai.expect(addSpy).to.have.been.calledTwice
+				m.chai.expect(removeSpy).to.not.have.been.called
+				m.chai.expect(addSpy).to.have.been.calledWith(foo: 'bar')
+				m.chai.expect(addSpy).to.have.been.calledWith(foo: 'baz')
+				scanner.stop()
+				done()
+			, 500
+
+	describe 'given two removals that happen at the same time', ->
+
+		beforeEach ->
+			@driveFinder = m.sinon.stub()
+			@driveFinder.onFirstCall().returns(Promise.resolve([ {foo: 'bar'}, {foo: 'baz'} ]))
```

Comment:
You can pass the additions as `options.drives` in the constructor to simplify the stub and avoid having to assert that `addSpy` was called twice (should have been called zero times).